pixano/apps/main.py

```diff
@@ -50,12 +51,31 @@ def create_app(settings: Settings = Settings()) -> FastAPI:
     model_dir = settings.data_dir / "models"
     model_dir.mkdir(exist_ok=True)
 
-    # Mount data directory (datasets + models)
-    app.mount(
-        "/data",
-        StaticFiles(directory=settings.data_dir),
-        name="data",
-    )
+    if not isinstance(settings.data_dir, S3Path):
+        # Mount data directory (datasets + models)
+        app.mount(
+            "/data",
+            StaticFiles(directory=settings.data_dir),
+            name="data",
+        )
+    else:
+        # don't need to mount dataset, but still need to mount model
+        if settings.local_model_dir is None:
+            # try to get model from S3 /models
+            # TODO
+            # list models in settings.data_dir / "models"
+            # dl them locally (where ??) and use this as mount point
+            # settings.local_model_dir = (settings.data_dir / "models").open()
+            raise Exception(
+                "download models from S3 not implemented yet, please set LOCAL_MODEL_DIR env var to a path with /models/<sam_model.onnx>)"
+            )
+        else:
+            # use local model
```

Comment:
Since downloading is not supported, maybe remove this and make local_model_dir mandatory in the `__init__` method of `Settings` when `scheme == "s3"` instead.